### PR TITLE
Book routes test

### PR DESCRIPTION
**Pull request by fcavazzoli**
## Testing Books endpoints

the users must be logged to access to the data of this endpoints.

### CODE REVIEW

File: .byebug_history [hunk 1]
```diff
@@ -0,0 +1,13 @@
+q
```
Comment (jhonmunoz) [inline .byebug_history]: remove this file from the repeo

File: spec/controllers/books_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        get :show, params: { id: book.id }
+      end
+
+      it 'responds with 200 status' do
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'responses with the book' do
+        expected = ActiveModel::Serializer::CollectionSerializer.new(book, each_serializer: BookSerializer).to_json
+        expect(response.body.to_json) =~ JSON.parse(expected)
```
Comment (jhonmunoz) [inline spec/controllers/books_controller_spec.rb]: why are you using `=~` to compare them? try usign a different approach

Comment (cquiro) [reply spec/controllers/books_controller_spec.rb]: Check out the docs:

https://www.rubydoc.info/github/rspec/rspec-expectations/RSpec%2FMatchers:equal

File: .byebug_history [hunk 3]
```diff
@@ -0,0 +1,9 @@
+c
```
Comment (floresv) [inline .byebug_history]: Remove this file.

File: app/assets/javascripts/books.coffee [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (floresv) [inline app/assets/javascripts/books.coffee]: Do you need this?

File: app/assets/stylesheets/books.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the books controller here.
```
Comment (floresv) [inline app/assets/stylesheets/books.scss]: Do you need this?

File: app/controllers/api/v1/books_controller.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class BooksController < ApplicationController
+      include Wor::Paginate
+      before_action :authenticate_user!
+      def index
+
```
Comment (floresv) [inline app/controllers/api/v1/books_controller.rb]: empty line, why?

File: app/controllers/api/v1/books_controller.rb [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+        @book = Book.all
+        render_paginated @book, each_serializer: BookSerializer
```
Comment (floresv) [inline app/controllers/api/v1/books_controller.rb]: Remember using worpaginate, https://github.com/Wolox/wor-paginate#basic-usage

File: app/models/rent.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class Rent < ApplicationRecord
+  belongs_to :user, inverse_of: false
```
Comment (floresv) [inline app/models/rent.rb]: this looks like belongs to another PR 🤔

File: app/models/user.rb [hunk 9]
```diff
@@ -8,5 +8,6 @@ class User < ApplicationRecord
 
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+  has_many :rents, dependent: :restrict_with_exception
```
Comment (floresv) [inline app/models/user.rb:8]: this looks like belongs to another PR 🤔

File: config/database.yml [hunk 10]
```diff
@@ -88,6 +88,10 @@ test:
...
-  database: app_production
-  username: app
-  password: <%= ENV['APP_DATABASE_PASSWORD'] %>
+  url: <%= ENV['DATABASE_URL'] %>
+  database: dadimm3p8dtdhr
+  username: ttuenhgjczzrmt
+  Port: 5432
+  password: f531d5d5d2437256fbff2009620a4b535f79d45192c84b30d46da6fc48d14e05
+  Host: ec2-54-83-36-37.compute-1.amazonaws.com
+  URI: postgres://ttuenhgjczzrmt:f531d5d5d2437256fbff2009620a4b535f79d45192c84b30d46da6fc48d14e05@ec2-54-83-36-37.compute-1.amazonaws.com:5432/dadimm3p8dtdhr
```
Comment (floresv) [inline config/database.yml]: Why you need this?

File: config/initializers/devise.rb [hunk 11]
```diff
@@ -0,0 +1,3 @@
+Devise.setup do |config|
+  config.secret_key = '6e2fd1bfe55c29d0ac32696bf93117642ba5dd3d7c926812958692a6a4453e98f3c93756650e834b1d5315497e1d1732acf7734621cbe41e62ac6e973e94eb68'
```
Comment (floresv) [inline config/initializers/devise.rb]: Use rails secrets, please.

File: config/routes.rb [hunk 12]
```diff
@@ -1,4 +1,11 @@
 Rails.application.routes.draw do
+  namespace 'api' do
+    namespace 'v1' do
+      resources :books
+      get 'api/v1/books/show', to: 'books#show', controller: 'books'
+      get 'api/v1/books/index', to: 'books#index', controller: 'books'
```
Comment (floresv) [inline config/routes.rb]: Prefer use of [resources](https://guides.rubyonrails.org/routing.html#configuring-the-rails-router)

File: spec/controllers/books_controller_spec.rb (hunk 2 above)
Comment (floresv) [reply spec/controllers/books_controller_spec.rb]: I agree with Carlos and Jhon

File: app/controllers/api/v1/books_controller.rb (hunk 6 above)
Comment (fcavazzoli) [reply app/controllers/api/v1/books_controller.rb]: byebug was there 🙈

File: app/assets/stylesheets/books.scss (hunk 5 above)
Comment (fcavazzoli) [reply app/assets/stylesheets/books.scss]: no I dont

File: app/models/user.rb (hunk 9 above)
Comment (fcavazzoli) [reply app/models/user.rb:8]: nope, model rents was implemeted before this pr

File: config/database.yml (hunk 10 above)
Comment (fcavazzoli) [reply config/database.yml]: it was to use heroku

File: config/routes.rb (hunk 12 above)
Comment (fcavazzoli) [reply config/routes.rb]: solved

**Comment by fcavazzoli** [comment]
I removed the test wich includes to compare equal paginated vs response cause the response was correctly paginates but i wasnt able to replicate an "expected" in the test file

File: .byebug_history (hunk 1 above)
Comment (floresv) [reply .byebug_history]: @fedecavazzoli  ?

File: app/controllers/api/v1/books_controller.rb [hunk 13]
```diff
@@ -7,12 +7,13 @@ class BooksController < ApplicationController
       before_action :authenticate_user!
       def index
         @book = Book.all
-        render_paginated @book, each_serializer: BooksSerializer
+        render_paginated @book, serializer: BookSerializer
```
Comment (floresv) [inline app/controllers/api/v1/books_controller.rb]: https://github.com/Wolox/wor-paginate#basic-usage
```suggestion
        render_paginated Book
```

File: config/environments/production.rb [hunk 14]
```diff
@@ -28,7 +28,7 @@
   # config.assets.css_compressor = :sass
 
   # Do not fallback to assets pipeline if a precompiled asset is missed.
-  config.assets.compile = false
+  config.assets.compile = true
```
Comment (floresv) [inline config/environments/production.rb]: Why?

File: config/initializers/devise_token_auth.rb [hunk 15]
```diff
@@ -19,7 +19,7 @@
   # time. In this case, each request in the batch will need to share the same
   # auth token. This setting determines how far apart the requests can be while
   # still using the same auth token.
-  # config.batch_request_buffer_throttle = 5.seconds
+  # config.batch_request_buffer_throttle = 1.day
```
Comment (floresv) [inline config/initializers/devise_token_auth.rb:22]: Why?

**Review by floresv — CHANGES_REQUESTED** [review]
Review it please.

File: config/environments/production.rb (hunk 14 above)
Comment (fcavazzoli) [reply config/environments/production.rb]: reading about heroku deploy i found this in google... i can remove it i think

File: config/initializers/devise_token_auth.rb (hunk 15 above)
Comment (fcavazzoli) [reply config/initializers/devise_token_auth.rb:22]: part of the big problen with autentication, it sets the time between new tokens of user, it is just to pass the tests, we had a really big problem with that.

**Review by floresv — APPROVED** [review]
LGTM
